videoCallingQuickstart/app/src/main/java/com/example/videocallingquickstart/MainActivity.java

```diff
@@ -0,0 +1,292 @@
+            acsException.printStackTrace();
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void turnOffLocalVideo() {
+        try {
+            ((LinearLayout) findViewById(R.id.localvideocontainer)).removeAllViews();
+            previewRenderer.dispose();
+            previewRenderer = null;
+            call.stopVideo(this, currentVideoStream).get();
+        } catch (CallingCommunicationException acsException) {
+            acsException.printStackTrace();
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void showPreview(LocalVideoStream stream) {
+        // Create renderer
+        previewRenderer = new VideoStreamRenderer(stream, this);
+        LinearLayout layout = ((LinearLayout)findViewById(R.id.localvideocontainer));
+        preview = previewRenderer.createView(new CreateViewOptions(ScalingMode.FIT));
+        runOnUiThread(() -> {
+            layout.addView(preview);
+        });
+    }
+
+    private void handleCallOnStateChanged(PropertyChangedEvent args) {
+        if (call.getState() == CallState.CONNECTED) {
+            handleInitialCallState();
+        }
+        if (call.getState() == CallState.DISCONNECTED) {
+            if (previewRenderer != null) {
+                previewRenderer.dispose();
+            }
+        }
+    }
+
+    private void handleInitialCallState() {
+        LinearLayout participantVideoContainer = findViewById(R.id.remotevideocontainer);
+        handleAddedParticipants(call.getRemoteParticipants(),participantVideoContainer);
+    }
+
+    private void answerIncomingCall() {
+        Context context = this.getApplicationContext();
+        if (incomingCall == null){
+            return;
+        }
+        AcceptCallOptions acceptCallOptions = new AcceptCallOptions();
+        VideoDeviceInfo camera = deviceManager.getCameras().get(0);
+        LocalVideoStream[] localVideoStreams = new LocalVideoStream[1];
+        localVideoStreams[0] = new LocalVideoStream(camera, context);
+        VideoOptions videoOptions = new VideoOptions(localVideoStreams);
+        acceptCallOptions.setVideoOptions(videoOptions);
+        showPreview(localVideoStreams[0]);
+        try {
+            call = incomingCall.accept(context, acceptCallOptions).get();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        }
+
+        call.addOnRemoteParticipantsUpdatedListener(this::handleRemoteParticipantsUpdate);
+        call.addOnStateChangedListener(this::handleCallOnStateChanged);
+    }
+
+    public void handleRemoteParticipantsUpdate(ParticipantsUpdatedEvent args) {
+        LinearLayout participantVideoContainer = findViewById(R.id.remotevideocontainer);
+        handleAddedParticipants(args.getAddedParticipants(),participantVideoContainer);
```

Comment:
What about participants that leave the call? How do we remove them from the UI?

Reply:
The call will end since it's 1:1 video calling.